borsh-derive-internal/src/struct_ser.rs

```diff
@@ -28,14 +32,6 @@ pub fn struct_ser(input: &ItemStruct, cratename: Ident) -> syn::Result<TokenStre
                     #cratename::BorshSerialize::serialize(&self.#field_name, writer)?;
                 };
                 body.extend(delta);
-
-                let field_type = &field.ty;
-                where_clause.predicates.push(
-                    syn::parse2(quote! {
-                        #field_type: #cratename::ser::BorshSerialize
-                    })
-                    .unwrap(),
-                );
             }
         }
         Fields::Unnamed(fields) => {
```

Comment:
strangely, there was no similar bound computation here for `UnnamedFields`
```
 #field_type: #cratename::ser::BorshSerialize
 ```
 Correspondingly, derive worked for simple recursive tuple struct (in contrast to struct with named fields):
 ```rust
    /// OK
     #[derive(Serialize, BorshSerialize)]
    struct CRecATuple (String, Box<CRecATuple>);
```
At the same time, derive for simple generic tuple struct didn't work:
```rust 
    /// ERROR: error[E0277]: the trait bound `T: BorshSerialize` is not satisfied
    #[derive(Serialize, BorshSerialize)]
    struct CRecATupleGeneric<T> (T, u32);

    /// OK
    #[derive(Serialize, BorshSerialize)]
    struct CRecANamedGeneric<T> {
        a: T,
        b: u32,
        
    }


```

Reply:
> At the same time, derive for simple generic tuple struct didn't work

This is the catch. Well, we traded one issue for another 😄 

It is fixed now, right?

Reply:
nope, it should be win-win in this situation, at least in the context of this comment